presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java

```diff
@@ -1290,4 +1295,22 @@ public FeaturesConfig setOptimizeNullsInJoin(boolean optimizeNullsInJoin)
         this.optimizeNullsInJoin = optimizeNullsInJoin;
         return this;
     }
+
+    public List<List<String>> getPartitionKeysToWarnOnNoFiltering()
```

Comment:
We usually keep the type from getter method consistent to the setter method. 

What about return `String` here and do the JSON parse in `WarnOnScanWithoutPartitionPredicate` ?

Reply:
Having a `List<List<>>` for configuration looks too complicated .. 

As discussed offline, let's focus on simple use case. The semantic can be: 
```
Here is a list of common partitioned columns.  If the table is partitioned on any of these columns, 
we expect the query need to have filter on at least one of those partitioned columns.
```

And in most companies in Bay Area, this will just be set to `[ds]` .